pkgs/jnigen/doc/threading.md

```diff
@@ -0,0 +1,46 @@
+## Threading considerations
+
+Unlike method channels, JNIgen uses FFI calls. This means that the calls happen
```

Comment:
If you expand on this explanation, I think it'll make the section below more clear. My understanding is that jnigen calls the Dart callback directly if they're already on the correct thread, and dispatches a message to the target isolate if not. And in the dispatching case the callback can be configured to wait for a response or not. So your documentation should mention each of those cases.

Also mention that, in any case, the Dart callback is always run in the isolate it was created in (this is something users have been confused about). Maybe also add a note that in upcoming versions of flutter, the main isolate will always run on the platform thread (I couldn't find any docs about that, but you could link to [this bug](https://github.com/flutter/flutter/issues/150525)).